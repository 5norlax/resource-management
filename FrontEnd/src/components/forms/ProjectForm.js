import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

const ProjectForm = ({ index, removeProject, ProjName, ProjDate, ProjAssociation, ProjInfo }) => {
    var today = new Date();
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projName">Name</Label>
                    <Input type="text" name="projName" id="projName" placeholder="Project Name" value={ProjName} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projDate">Date</Label>
                    <Input type="number" min="1900" max={today.getFullYear()} step="1" defaultValue={today.getFullYear()} value={ProjDate} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projAssociation">Association</Label>
                    <Input type="text" name="projAssociation" id="projAssociation" placeholder="Project Association" value={ProjAssociation} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projDetails1">Details</Label>
                    <Input type="textarea" name="projDetails1" id="projDetails1" placeholder="Project Details" value={ProjInfo} />
                </FormGroup>
                <FormGroup className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => {removeProject(index)}} style={{ marginTop: '32px' }} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default ProjectForm