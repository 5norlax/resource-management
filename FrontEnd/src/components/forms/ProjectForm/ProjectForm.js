import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

export default React.memo(function ProjectForm({ index, removeProject, ProjName, ProjDate, ProjAssociation, ProjInfo, handleProjectChange }) {

    var today = new Date();

    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="ProjName">Name</Label>
                    <Input required type="text" name="ProjName" id="ProjName" placeholder="Project Name" value={ProjName} onChange={(e) => { handleProjectChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="ProjDate">Year</Label>
                    <Input required type="number" name="ProjDate" id="ProjDate" min="1900" max={today.getFullYear()} step="1" placeholder="Project Date" value={ProjDate} onChange={(e) => { handleProjectChange(index, e) }} data-testid="projectDateInput"/>
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="ProjAssociation">Association</Label>
                    <Input required type="text" name="ProjAssociation" id="ProjAssociation" placeholder="Project Association" value={ProjAssociation} onChange={(e) => { handleProjectChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="ProjInfo">Details</Label>
                    <Input type="textarea" name="ProjInfo" id="ProjInfo" placeholder="Project Details" value={ProjInfo} onChange={(e) => { handleProjectChange(index, e) }} />
                </FormGroup>
                <FormGroup className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12 text-right">
                    <Button onClick={() => { removeProject(index) }} style={{ marginTop: '32px' }} color="danger">Remove Project</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
});