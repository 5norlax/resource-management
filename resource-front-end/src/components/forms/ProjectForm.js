import React from 'react';
import { Label, Input, FormGroup, Row, Button } from 'reactstrap';

const ProjectForm = ({index, removeProject}) => {
    return (
        <div>
            <Row>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projName">Name</Label>
                    <Input type="text" name="projName" id="projName" placeholder="Project Name" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projDate">Date</Label>
                    <Input type="text" name="projDate" id="projDate" placeholder="Project Date" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projAssociation">Association</Label>
                    <Input type="text" name="projAssociation" id="projAssociation" placeholder="Project Association" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projDetails1">Details</Label>
                    <Input type="textarea" name="projDetails1" id="projDetails1" placeholder="Project Details" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projDetails2">Details</Label>
                    <Input type="textarea" name="projDetails2" id="projDetails2" placeholder="Project Details" />
                </FormGroup>
                <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                    <Label for="projDetails3">Details</Label>
                    <Input type="textarea" name="projDetails3" id="projDetails3" placeholder="Project Details" />
                </FormGroup>
                <FormGroup className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12">
                    <Button onClick={() => {removeProject(index)}} style={{}} color="danger">Remove Education</Button>
                </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default ProjectForm