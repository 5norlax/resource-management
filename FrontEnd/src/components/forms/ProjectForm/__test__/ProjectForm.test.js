import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ProjectForm from '../ProjectForm';

afterEach(cleanup);

describe('Test ProjectForm', () => {

  it('matches snapshot', () => {
    const tree = renderer.create(<ProjectForm />).toJSON()
    expect(tree).toMatchSnapshot();
  });

});