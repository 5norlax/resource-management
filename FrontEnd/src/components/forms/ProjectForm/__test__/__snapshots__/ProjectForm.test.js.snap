// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProjectForm matches snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="ProjName"
      >
        Name
      </label>
      <input
        className="form-control"
        id="ProjName"
        name="ProjName"
        onChange={[Function]}
        placeholder="Project Name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="ProjDate"
      >
        Year
      </label>
      <input
        className="form-control"
        id="ProjDate"
        max={2020}
        min="1900"
        name="ProjDate"
        onChange={[Function]}
        placeholder="Project Date"
        required={true}
        step="1"
        type="number"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="ProjAssociation"
      >
        Association
      </label>
      <input
        className="form-control"
        id="ProjAssociation"
        name="ProjAssociation"
        onChange={[Function]}
        placeholder="Project Association"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <label
        className=""
        htmlFor="ProjInfo"
      >
        Details
      </label>
      <textarea
        className="form-control"
        id="ProjInfo"
        name="ProjInfo"
        onChange={[Function]}
        placeholder="Project Details"
      />
    </div>
    <div
      className="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-xs-12 text-right form-group"
    >
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        style={
          Object {
            "marginTop": "32px",
          }
        }
        type="button"
      >
        Remove Project
      </button>
    </div>
  </div>
  <hr />
</div>
`;
